### PR TITLE
Fixed some bugs in ExplicitSSPRungeKutta

### PR DESCRIPTION
**Pull request by joshuahansel**
This solves https://github.com/idaholab/moose/issues/18646 and https://github.com/idaholab/moose/issues/18643.

### CODE REVIEW

**Comment by joshuahansel** [comment]
@andrsd

File: framework/src/timeintegrators/ExplicitSSPRungeKutta.C [hunk 1]
```diff
@@ -89,10 +97,6 @@ ExplicitSSPRungeKutta::computeADTimeDerivatives(ADReal & ad_u_dot,
     ad_u_dot = std::numeric_limits<typename ADReal::value_type>::quiet_NaN();
     return;
```
Comment (lindsayad) [inline framework/src/timeintegrators/ExplicitSSPRungeKutta.C:98]: Can remove this no-longer-early-return now

File: framework/src/timeintegrators/ExplicitTimeIntegrator.C [hunk 2]
```diff
@@ -63,6 +64,9 @@ ExplicitTimeIntegrator::initialSetup()
 void
 ExplicitTimeIntegrator::init()
 {
+  if (_fe_problem.solverParams()._type != Moose::ST_LINEAR)
+    mooseError(
+        "The chosen time integrator requires 'solve_type = LINEAR' in the Executioner block.");
```
Comment (lindsayad) [inline framework/src/timeintegrators/ExplicitTimeIntegrator.C:69]: Love it!

File: test/tests/time_integrators/explicit_ssp_runge_kutta/tests [hunk 3]
```diff
@@ -32,4 +32,15 @@
...
   []
+
+  [invalid_solve_type]
+    type = RunException
+    input = 'explicit_ssp_runge_kutta.i'
+    cli_args = "Executioner/solve_type=NEWTON"
+    expect_err = "The chosen time integrator requires 'solve_type = LINEAR'"
+
+    requirement = "The system shall provide an informative error message when "
+      "a user uses an invalid solve type to an explicit time integrator."
```
Comment (lindsayad) [inline test/tests/time_integrators/explicit_ssp_runge_kutta/tests:43]:
```suggestion
      "a user uses an invalid solve type for an explicit time integrator."
```

**Review by lindsayad — CHANGES_REQUESTED** [review]

**Review by lindsayad — APPROVED** [review]

**Pull request merged**